homeassistant/components/here_weather/__init__.py

```diff
@@ -0,0 +1,187 @@
+
+
+async def async_unload_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+    unload_ok = all(
+        await asyncio.gather(
+            *[
+                hass.config_entries.async_forward_entry_unload(config_entry, component)
+                for component in PLATFORMS
+            ]
+        )
+    )
+    if unload_ok:
+        hass.data[DOMAIN][config_entry.entry_id].unsub_handler()
+        hass.data[DOMAIN].pop(config_entry.entry_id)
+
+    return unload_ok
+
+
+class HEREWeatherData:
+    """Get the latest data from HERE."""
+
+    def __init__(self, hass: HomeAssistant, config_entry: ConfigEntry) -> None:
+        """Initialize the data object."""
+        self.hass = hass
+        self.config_entry = config_entry
+        self.here_client = herepy.DestinationWeatherApi(config_entry.data[CONF_API_KEY])
+        self.latitude = config_entry.data.get(CONF_LATITUDE)
+        self.longitude = config_entry.data.get(CONF_LONGITUDE)
+        self.location_name = config_entry.data.get(CONF_LOCATION_NAME)
+        self.zip_code = config_entry.data.get(CONF_ZIP_CODE)
+        self.weather_product_type = herepy.WeatherProductType[
+            config_entry.data[CONF_MODE]
+        ]
+        self.units = config_entry.data.get(CONF_UNIT_SYSTEM)
+        self.coordinator = None
+        self.unsub_handler = None
+
+    async def async_setup(self):
+        """Set up the here_weather integration."""
+        self.add_options()
+        self.unsub_handler = self.config_entry.add_update_listener(
+            self.async_options_updated
+        )
+        await self._async_create_coordinator()
+        return True
+
+    def add_options(self):
+        """Add options for here_weather integration."""
+        if not self.config_entry.options:
+            options = {CONF_SCAN_INTERVAL: DEFAULT_SCAN_INTERVAL}
+            self.hass.config_entries.async_update_entry(
+                self.config_entry, options=options
+            )
+
+    async def _async_create_coordinator(self):
+        """Create or recreate the DataUpdateCoordinator."""
+        self.coordinator = DataUpdateCoordinator(
+            self.hass,
+            _LOGGER,
+            name=DOMAIN,
+            update_method=self.async_update,
+            update_interval=timedelta(
+                seconds=self.config_entry.options[CONF_SCAN_INTERVAL]
+            ),
+        )
+        await self.coordinator.async_refresh()
+
+    async def async_update(self) -> None:
+        """Handle data update with the DataUpdateCoordinator."""
+        try:
+            async with async_timeout.timeout(10):
+                return await self.hass.async_add_executor_job(self._get_data)
+        except herepy.InvalidRequestError as error:
+            raise UpdateFailed(f"Unable to fetch data from HERE: {error.message}")
+
+    def _get_data(self):
+        """Get the latest data from HERE."""
+        is_metric = convert_units_to_boolean(self.units)
```

Comment:
What's the purpose of `convert_units_to_boolean()`? Running `is_metric = self.units == CONF_UNIT_SYSTEM_METRIC` is simple enough.